Modules/Module_Processing/Module_ASL/xASL_wrp_ResampleASL.m

```diff
@@ -245,24 +245,29 @@ function xASL_wrp_ResampleASL(x)
 MaskIM = MaskIM>(0.7*max(MaskIM(:)));
 
 if xASL_exist(x.P.Pop_Path_PWI4D)
-    [PWI4D, json] = xASL_io_Nifti2Im(x.P.Pop_Path_PWI4D);
+    [PWI4D, json_PWI4D] = xASL_io_Nifti2Im(x.P.Pop_Path_PWI4D);
+	% Convert JSON from BIDS to Legacy
+	json_PWI4D = xASL_bids_parms2BIDS([], json_PWI4D, 0);
     nVolumesPWI4D = size(PWI4D, 4);
 else
     nVolumesPWI4D = 1;
+	json_PWI4D = [];
 end
 
 if nVolumesPWI4D>3
-    if isempty(json)
+    if isempty(json_PWI4D)
         warning(['Reverting to x.Q memory, json file missing: ' x.P.Pop_Path_PWI4D_json]);
-        json.Q = x.Q;
+        json_PWI4D.Q = x.Q;
     end
     
-    %% PM: here we need to use the positions for the earliest echo, and the latest PLD-labdur
-    
-    % PWI4D_statsIndices = json.EchoTime_PWI4D==min(x.Q.uniqueEchoTime) & json.InitialPLD_PWI4D==max(x.Q.uniqueInitial_PLD);
-    PWI4D_statsIndices = json.Q.Initial_PLD==max(x.Q.uniqueInitial_PLD);
-    %% #1543 For some reason this didn't work for the combination shortest TE & longest PLD
-    %% Better to repeat this for every TE-PLD-LD combination 
+    %% Use the positions for the earliest echo, and the latest PLD-labdur

```

Comment:
I thought we did this with the TE-PLD-LD weighting function that I wrote?

I am confused why we don't also use this here

Reply:
We did weighted for registration. This is for single-PLD quantification of multi-PLD data. For that we can't weight

Reply:
Are you sure we can't weight? But more importantly: always choosing the last PLD will not necessarily be the best option for single-PLD quantification of multi-PLD data, as you pointed out before. Shall we make a new issue out of this (with low priority), then I agree with you here.

Reply:
For quantification, we need to know the exact parameters. Doing a single-PLD quantification on averaged multi-PLD data is a bad idea, so we really have to choose a single-PLD or do a proper multi-PLD quantification rather than weighted averaging.

Choosing the last PLD is indeed a bad idea, that's why I rewrote the code to choose the PLD closest to 2000 (which was our top choice for the weighting as well). 

I'm fine choosing a slightly different strategy than 2000ms. Any suggestions? Cause very likely, the solution will be very simple and we can make the change immediately without making a new issue.

Reply:
I'm of course thinking of other users and other sequences. The PWI we are choosing here is used to calculate SD and SNR, not for visualization. So it makes no sense to do weighting as this mess up things.